test/shared/src/main/scala/zio/test/TestTimeoutException.scala

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright 2019 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.test
+
+final case class TestTimeoutException(message: String) extends Throwable(message, null, true, false)
```

Comment:
Maybe make this extend `TimeoutException`?

Reply:
It was the case initially but we changed for 2 reasons:
- not generating any unnecessary call stack
- since we don't display the full cause in `DefautTestReporter` for this error type, if a user uses `TimeoutException` in their code, they won't see the cause either. By using a separate class we can mute just that one.

Reply:
If we made `TestTimeoutException` extend `TimeoutException` and pattern matched on `TestTimeoutException` in `DefaultTestReporter` wouldn't that still get us what we want? Or do we lose the ability to specify no stack trace if we do that?

Reply:
I think it would generate a stack trace then. But what would be the benefit?

Reply:
Yeah, I thought it might just be good for consistency if we ever want to do anything else with to handle that exception, but we will always know what it is. I'm good to merge.